pkg/microservice/aslan/core/workflow/service/workflow/workflow_task.go

```diff
@@ -702,6 +711,16 @@ func CreateWorkflowTask(args *commonmodels.WorkflowTaskArgs, taskCreator string,
 			AddSubtaskToStage(&stages, stask, target.Name+"_"+target.ServiceName)
 		}
 	}
+	for _, test := range args.Tests {
+		for _, build := range test.Builds {
+			repoInfo, err := systemconfig.New().GetCodeHost(build.CodehostID)
+			if err != nil {
+				log.Errorf("Failed to get proxy settings for codehost ID: %d, the error is: %s", build.CodehostID, err)
+				return nil, err
+			}
+			build.EnableProxy = repoInfo.EnableProxy
+		}
+	}
```

Comment:
what if we build a map of codehosts with one query operation to avoid reading codehost config from db for every build info in both test and build stages?

Reply:
This shall be done in the next patch